src/module/chat-message/listeners/cards.ts

```diff
@@ -70,7 +70,45 @@ export const ChatCards = {
                 else if (action === "spellDamage") spell?.rollDamage(event);
                 else if (action === "spellCounteract") spell?.rollCounteract(event);
                 else if (action === "spellTemplate") spell?.placeTemplate();
-                else if (action === "selectVariant") (await spell?.variantPrompt())?.toMessage();
+                else if (action === "selectVariant") {
+                    const spellLvl = Number($html.find<HTMLDivElement>("div.chat-card").attr("data-spell-lvl")) || 1;
+                    const overlayIdString = $button.attr("data-overlay-ids");
+                    const originalId = $button.attr("data-original-id") ?? "";
+                    if (overlayIdString) {
+                        const overlayIds = overlayIdString.split(",").map((id) => id.trim());
+                        const variantSpell = spell?.loadVariant({ overlayIds, castLevel: spellLvl });
+                        if (variantSpell) {
+                            const variantMessage = await variantSpell.toMessage(undefined, {
+                                create: false,
+                                data: { spellLvl },
+                            });
+                            if (variantMessage) {
+                                const messageSource = variantMessage.toObject();
+                                messageSource.flags = mergeObject(
```

Comment:
Is this not already handled by Spell#toMessage()? It injects some flags if its a variant.

Reply:
It wasn't working without it and I forgot to investigate further. I'll take another look.